effect/shared/src/main/scala/com/monovore/decline/effect/CommandIOApp.scala

```diff
@@ -14,34 +16,55 @@ abstract class CommandIOApp(
 
   def main: Opts[IO[ExitCode]]
 
-  private[this] def command: Command[IO[ExitCode]] = {
-    val showVersion = {
-      if (version.isEmpty) Opts.never
-      else {
-        val flag = Opts.flag(
-          long = "version",
-          short = "v",
-          help = "Print the version number and exit.",
-          visibility = Visibility.Partial
-        )
-        flag.as(IO(System.out.println(version)).as(ExitCode.Success))
-      }
-    }
+  override final def run(args: List[String]): IO[ExitCode] = {
+    import CommandIOApp._
+
+    val opts = if (version.isEmpty) main else addVersionFlag(main)(version)
```

Comment:
why not call `CommandIOApp.run[IO](...)` here and remove the duplication?

Reply:
The only little difference is the version parameter, being an Option in `CommandIOApp.run`, but it could be easily adjusted

Reply:
Pushed the removal of duplication 🙂